hoomd/md/PotentialPairAlchemical.h

```diff
@@ -0,0 +1,719 @@
+                // (FLOPS: 8)
+                fi += dx * force_divr;
+                pei += pair_eng * Scalar(0.5);
+                if (compute_virial)
+                    {
+                    virialxxi += force_div2r * dx.x * dx.x;
+                    virialxyi += force_div2r * dx.x * dx.y;
+                    virialxzi += force_div2r * dx.x * dx.z;
+                    virialyyi += force_div2r * dx.y * dx.y;
+                    virialyzi += force_div2r * dx.y * dx.z;
+                    virialzzi += force_div2r * dx.z * dx.z;
+                    }
+
+                // add the force to particle j if we are using the third law (MEM TRANSFER: 10
+                // scalars / FLOPS: 8) only add force to local particles
+                if (third_law && j < m_pdata->getN())
+                    {
+                    unsigned int mem_idx = j;
+                    h_force.data[mem_idx].x -= dx.x * force_divr;
+                    h_force.data[mem_idx].y -= dx.y * force_divr;
+                    h_force.data[mem_idx].z -= dx.z * force_divr;
+                    h_force.data[mem_idx].w += pair_eng * Scalar(0.5);
+                    if (compute_virial)
+                        {
+                        h_virial.data[0 * m_virial_pitch + mem_idx] += force_div2r * dx.x * dx.x;
+                        h_virial.data[1 * m_virial_pitch + mem_idx] += force_div2r * dx.x * dx.y;
+                        h_virial.data[2 * m_virial_pitch + mem_idx] += force_div2r * dx.x * dx.z;
+                        h_virial.data[3 * m_virial_pitch + mem_idx] += force_div2r * dx.y * dx.y;
+                        h_virial.data[4 * m_virial_pitch + mem_idx] += force_div2r * dx.y * dx.z;
+                        h_virial.data[5 * m_virial_pitch + mem_idx] += force_div2r * dx.z * dx.z;
+                        }
+                    }
+                }
+            }
+
+        // finally, increment the force, potential energy and virial for particle i
+        unsigned int mem_idx = i;
+        h_force.data[mem_idx].x += fi.x;
+        h_force.data[mem_idx].y += fi.y;
+        h_force.data[mem_idx].z += fi.z;
+        h_force.data[mem_idx].w += pei;
+        if (compute_virial)
+            {
+            h_virial.data[0 * m_virial_pitch + mem_idx] += virialxxi;
+            h_virial.data[1 * m_virial_pitch + mem_idx] += virialxyi;
+            h_virial.data[2 * m_virial_pitch + mem_idx] += virialxzi;
+            h_virial.data[3 * m_virial_pitch + mem_idx] += virialyyi;
+            h_virial.data[4 * m_virial_pitch + mem_idx] += virialyzi;
+            h_virial.data[5 * m_virial_pitch + mem_idx] += virialzzi;
+            }
+        }
+    pkgFinalize(pkg);
+
+    computeTailCorrection();
+    }
+
+template<class evaluator, typename extra_pkg, typename alpha_particle_type>
+template<class InputIterator>
+inline void
+PotentialPairAlchemical<evaluator, extra_pkg, alpha_particle_type>::computeEnergyBetweenSets(
```

Comment:
Is there a reason `computeEnergyBetweenSets` includes the alchemical forces? Because if not we could delete this and rely on the implementation in `PotentialPair.h`.

Reply:
I don't know why the original author put that there. We can probably remove the method. I think that the new method likely isn't called as the subclass has different template arguments than the base and there is no non-templated base to define the virtual method signature.

Reply:
Removed.